rust/src/message/sum.rs

```diff
@@ -61,16 +61,28 @@ impl<B: AsRef<[u8]> + AsMut<[u8]>> SumMessageBuffer<B> {
 
 #[derive(Clone, Debug, PartialEq)]
 /// Encryption and decryption of sum messages.
-pub struct SumMessage<K, C, S, E> {
+pub struct SumMessage<K, C, S, E>
+where
+    K: Borrow<SumParticipantPublicKey>,
```

Comment:
I think it makes sense to have the trait bounds at the struct level, because it makes the compiler errors clearer when trying to create structs with inappropriate types.

Reply:
i agree. it makes calls to the fields a bit more verbose, like `self.pk.borrow().as_ref()` instead of `self.pk.as_ref()`, but that's definitely worth it.